### PR TITLE
fix: reset reserve only when neighborhood hopping

### PR DESCRIPTION
**Pull request by istae**
### Checklist

- [ ] I have read the [coding guide](https://github.com/ethersphere/bee/blob/master/CODING.md).
- [ ] My change requires a documentation update, and I have done it.
- [ ] I have added tests to cover my changes.
- [ ] I have filled out the description and linked the related issues.

### Description
also renamed "namespace" in chunkstamp and stampindex to "scope"
also cleaned up some unnecessary index store funcs

### CODE REVIEW

File: pkg/node/node.go [hunk 1]
```diff
@@ -293,21 +294,19 @@ func NewBee(
 			}
 
 			if nonceExists {
-				logger.Info("Override nonce %x to %x and clean state for neighborhood %s", nonce, newNonce, targetNeighborhood)
+				logger.Info("Override nonce and clean state for neighborhood", "old_none", hex.EncodeToString(nonce), "new_nonce", hex.EncodeToString(newNonce))
```
Comment (acha-bill) [inline pkg/node/node.go:297]:
```suggestion
				logger.Info("Override nonce and clean state for neighborhood", "old_nonce", hex.EncodeToString(nonce), "new_nonce", hex.EncodeToString(newNonce))
```

**Review by acha-bill — APPROVED** [review]

**Review by martinconic — APPROVED** [review]

File: pkg/storer/internal/reserve/reserve_test.go [hunk 2]
```diff
@@ -752,6 +753,85 @@ func TestIterate(t *testing.T) {
...
+			chs = append(chs, ch)
+		}
+	}
+
+	err = r.Reset(context.Background())
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	c, err := ts.IndexStore().Count(&reserve.BatchRadiusItem{})
```
Comment (nugaon) [inline pkg/storer/internal/reserve/reserve_test.go:802]: it could be checked before reset as well to show the expected count

**Review by nugaon — COMMENTED** [review]
I thought it keeps the pinned chunks alive somehow, or will that be implemented in a different PR?

File: pkg/storer/internal/reserve/reserve.go [hunk 3]
```diff
@@ -516,6 +516,99 @@ func (r *Reserve) IterateChunksItems(startBin uint8, cb func(*ChunkBinItem) (boo
 	return err
 }
 
+func (r *Reserve) Reset(ctx context.Context) error {
+
+	size := r.Size()
+
+	bRitems := make([]*BatchRadiusItem, 0, size)
```
[the remark this answers is not in the document]
Comment (istae) [reply pkg/storer/internal/reserve/reserve.go:523]: The size here is the actual number of chunks in the reserve. You are confusing it with reserve capacity.

File: pkg/storer/internal/reserve/reserve.go [hunk 4]
```diff
@@ -516,6 +516,99 @@ func (r *Reserve) IterateChunksItems(startBin uint8, cb func(*ChunkBinItem) (boo
...
+		}(item)
+	}
+
+	err = eg.Wait()
+	if err != nil {
+		return err
+	}
+
+	citems := make([]*chunkstamp.Item, 0, size)
+	err = r.st.IndexStore().Iterate(storage.Query{
```
[the remark this answers is not in the document]
Comment (istae) [reply pkg/storer/internal/reserve/reserve.go:583]: we can actually delete this one inside the other loop

File: pkg/storer/internal/reserve/reserve_test.go (hunk 2 above)
Comment (istae) [reply pkg/storer/internal/reserve/reserve_test.go:802]: yes, good idea

File: pkg/storer/internal/reserve/reserve.go [hunk 5]
```diff
@@ -516,6 +516,99 @@ func (r *Reserve) IterateChunksItems(startBin uint8, cb func(*ChunkBinItem) (boo
...
+					)
+				})
+			})
+		}(item)
+	}
+
+	err = eg.Wait()
+	if err != nil {
+		return err
+	}
```
[the remark this answers is not in the document]
Comment (istae) [reply pkg/storer/internal/reserve/reserve.go:555]: i can assign nil to the slices after using them.

File: pkg/storer/internal/reserve/reserve.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (nugaon) [reply pkg/storer/internal/reserve/reserve.go:521]: nice!
is append better than just make with the length of `r.Size()` and assign by indices?

**Review by nugaon — APPROVED** [review]

**Review by martinconic — APPROVED** [review]

**Review by acha-bill — APPROVED** [review]

**Pull request merged**